Yehor/reactNetflix/src/redux/sagas/moviesSaga.ts

```diff
@@ -1,95 +1,88 @@
 import { takeLatest, call, put, all } from "redux-saga/effects";
+import { moviesReceived, error, loadData } from "../reducers/reducerMovies";
+import IMovie from "../../interfaces/IMovie";
 import {
-  MoviesTypes,
-  ILoadData,
-  receivedData,
-  error
-} from "../actions/moviesActions";
-import {
-  MovieTypes,
-  ICurrentMovieLoad,
+  currentMovieLoad,
   currentMovieReceived,
-  currentMovieError
-} from "../actions/movieActions";
-import IMovie from "../../interfaces/IMovie";
+  currentMovieError,
+} from "../reducers/reducerMovie";
 
-export const fetchMoviesApi = async (sortBy?: string, searchBy?: string, search?: string, offset?: number): Promise<IMovie[]> => {
-  const limit = 'limit=9'
-  
-  let url = 
-  searchBy && search && sortBy ? `https://reactjs-cdp.herokuapp.com/movies?sortBy=${sortBy}&sortOrder=desc&search=${search}&searchBy=${searchBy}&offset=${offset}&${limit}`:
-  searchBy && search ? `https://reactjs-cdp.herokuapp.com/movies?search=${search}&searchBy=${searchBy}&offset=${offset}&${limit}`:
-  sortBy ? `https://reactjs-cdp.herokuapp.com/movies?sortBy=${sortBy}&sortOrder=desc&offset=${offset}&${limit}`:
-  `https://reactjs-cdp.herokuapp.com/movies?sortBy=release_data&sortOrder=desc&offset=${offset}&${limit}`
+export const fetchMoviesApi = async (
+  sortBy?: string,
+  searchBy?: string,
+  search?: string,
+  offset?: number
+): Promise<IMovie[]> => {
+  const limit = 9;
+  const url =
+    searchBy && search
+      ? `https://reactjs-cdp.herokuapp.com/movies?sortBy=
+  ${sortBy ? sortBy : "release_data"}
+  &search=${search}&searchBy=${searchBy}&offset=${offset}&limit=${limit}`
+      : `https://reactjs-cdp.herokuapp.com/movies?sortBy=release_data&sortOrder=desc&offset=${offset}&limit=${limit}`;
   const getMovies = await fetch(url);
   const movies = await getMovies.json();
   return movies.data;
 };
 
-export const fetchCurrentMovieApi = async (id: string): Promise<void> => {
-  const getMovies = await fetch(
-    `http://reactjs-cdp.herokuapp.com/movies/${id}`
-  );
-  const movie = await getMovies.json();
-  console.log(movie);
-  return movie;
-};
+interface ILoadDataAction {
```

Comment:
don't like that you had to create an interface just to use it in saga. Seems redundant to me. My proposal is to use this instead:
ReturnType<typeof loadData >